File: homework/home5.1/Source/MegaData.java
package org.example;


public class MegaData {
    public static final int SMALL_ARRAY_SIZE = 1024;
    public static final int BIG_ARRAY_SIZE = 1024 * 1024;

    public final float[] smallArray;
    public final double[] bigArray;

    public MegaData() {
        smallArray = new float[SMALL_ARRAY_SIZE];
        bigArray = new double[BIG_ARRAY_SIZE];
        for (int i = 0; i < SMALL_ARRAY_SIZE; i++) {
            smallArray[i] = 42.0f;
        }
        for (int i = 0; i < BIG_ARRAY_SIZE; i++) {
            bigArray[i] = 42.0;
        }
    }

    public void reset() {
        for (int i = 0; i < SMALL_ARRAY_SIZE; i++) {
            smallArray[i] = 42.0f;
        }
        for (int i = 0; i < BIG_ARRAY_SIZE; i++) {
            bigArray[i] = 42.0;
        }
    }

}

File: classwork/classwork5.1/MegaData.cpp
#include <array>
#include <vector>
#include <algorithm>
#include <stdexcept>
#include <memory>

class MegaData {
public:
    static constexpr size_t SMALL_ARRAY_SIZE = 1024;
    static constexpr size_t BIG_ARRAY_SIZE = 1024 * 1024;

    MegaData() : smallArray(std::make_unique<float[]>(SMALL_ARRAY_SIZE)),
                 bigArray(std::make_unique<double[]>(BIG_ARRAY_SIZE)) {
        std::fill(smallArray.get(), smallArray.get() + SMALL_ARRAY_SIZE, 42.0f);
        std::fill(bigArray.get(), bigArray.get() + BIG_ARRAY_SIZE, 42.0);
    }

    void reset() {
        std::fill(smallArray.get(), smallArray.get() + SMALL_ARRAY_SIZE, 42.0f);
        std::fill(bigArray.get(), bigArray.get() + BIG_ARRAY_SIZE, 42.0);
    }

    bool operator==(const MegaData& other) const {
        return std::equal(smallArray.get(), smallArray.get() + SMALL_ARRAY_SIZE, other.smallArray.get()) &&
               std::equal(bigArray.get(), bigArray.get() + BIG_ARRAY_SIZE, other.bigArray.get());
    }

    std::unique_ptr<float[]> smallArray;
    std::unique_ptr<double[]> bigArray;
};